components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java

```diff
@@ -46,722 +39,728 @@
-	 *
-	 * @param onCacheHitRef reference to the onCacheHit sequence to be set.
-	 */
-	public void setOnCacheHitRef(String onCacheHitRef) {
-		this.onCacheHitRef = onCacheHitRef;
-	}
-
-	/**
-	 * This method gives the maximum size of the messages to be cached in bytes.
-	 *
-	 * @return maximum size of the messages to be cached in bytes.
-	 */
-	public int getMaxMessageSize() {
-		return maxMessageSize;
-	}
-
-	/**
-	 * This method sets the maximum size of the messages to be cached in bytes.
-	 *
-	 * @param maxMessageSize maximum size of the messages to be set in bytes.
-	 */
-	public void setMaxMessageSize(int maxMessageSize) {
-		this.maxMessageSize = maxMessageSize;
-	}
-
-	@Override
-	public Mediator getInlineSequence(SynapseConfiguration synCfg, int inlinedSeqIdentifier) {
-		if (inlinedSeqIdentifier == 0) {
-			if (onCacheHitSequence != null) {
-				return onCacheHitSequence;
-			} else if (onCacheHitRef != null) {
-				return synCfg.getSequence(onCacheHitRef);
-			}
-		}
-		return null;
-	}
-
+    /**
+     * The value of json content type as it appears in HTTP Content-Type header.
+     */
+    private final String jsonContentType = "application/json";
+    /**
+     * Cache configuration ID.
+     */
+    private String id;
+
+    /**
+     * The time duration for which the cache is kept.
+     */
+    private long timeout = CachingConstants.DEFAULT_TIMEOUT;
+
+    /**
+     * This specifies whether the mediator should be in the incoming path (to check the request) or in the outgoing path
+     * (to cache the response).
+     */
+    private boolean collector = false;
+
+    /**
+     * The SequenceMediator to the onCacheHit sequence to be executed when an incoming message is identified as an
+     * equivalent to a previously received message based on the value defined for the Hash Generator field.
+     */
+    private SequenceMediator onCacheHitSequence = null;
+
+    /**
+     * The reference to the onCacheHit sequence to be executed when an incoming message is identified as an equivalent
+     * to a previously received message based on the value defined for the Hash Generator field.
+     */
+    private String onCacheHitRef = null;
+
+    /**
+     * The headers to exclude when caching.
+     */
+    private String[] headersToExcludeInHash = {""};
+
+    /**
+     * This is used to define the logic used by the mediator to evaluate the hash values of incoming messages.
+     */
+    private DigestGenerator digestGenerator = CachingConstants.DEFAULT_HASH_GENERATOR;
+
+    /**
+     * The size of the messages to be cached in memory. If this is 0 then no disk cache, and if there is no size
+     * specified in the config  factory will assign a default value to enable disk based caching.
+     */
+    private int inMemoryCacheSize = -1;
+
+    /**
+     * Variable to represent 'NO_ENTITY_BODY' property of synapse.
+     */
+    private static final String NO_ENTITY_BODY = "NO_ENTITY_BODY";
```

Comment:
Why have we redefined this property here?
use org.apache.synapse.transport.passthru.PassThroughConstants#NO_ENTITY_BODY

Reply:
Fixed in #890